compiler/rustc_target/src/spec/json.rs

```diff
@@ -111,6 +111,23 @@ impl Target {
                     base.$key_name = Some(s.into());
                 }
             } );
+            ($key_name:ident, Option<Align>) => ( {
+                let name = (stringify!($key_name)).replace("_", "-");
+                if let Some(b) = obj.remove(&name).and_then(|b| b.as_u64()) {
+                    match Align::from_bits(b) {
+                        Ok(align) => base.$key_name = Some(align),
+                        Err(AlignFromBytesError::NotPowerOfTwo(align)) => {
+                            return Err(format!("`{align}` bits is not a valid value for \
+                                               minimum-global-alignment: not a power of 2 \
+                                               number of bytes"));
+                        }
+                        Err(AlignFromBytesError::TooLarge(align)) => {
+                            return Err(format!("`{align}` bits is not a valid value for \
+                                               minimum-global-alignment: too large"));
+                        }
+                    }
+                }
+            } );
```

Comment:
Please either move this outside the macro or fix this so it can actually apply to different keys.

Reply:
Fixed. I pattern matched on some of the other fields, but it is much more likely that the align type will get re-used for other keys of course.